### DISCUSSION TITLE
Ability to embed Zed buffer editor in native applications

### DISCUSSION THREAD

**Discussion by leonbreedt-sap** [zed-industries/zed#8805] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I am working on a SwiftUI native macOS application that needs some first class JavaScript/TypeScript editing. Usable SwiftUI components are a bit thin on the ground for this, so I have resorted to embedding Monaco from VSCode in my application, but I pay a heavy price for this (cold start of text editing is measured in seconds).

I would dearly love to be able to embed a feature-rich text editor component into my application, with some decent performance, like Zed.

Is there any universe in which:

1. This isn't an incredibly stupid idea.
2. This would be technically feasible with your architecture.
3. You would consider an embedding use case like this, where I can give out some kind of handle and screen coordinates/area to Zed, drop in a Zed buffer there, and feed it the data to be edited and get notified on change?

Thanks!

**Comment by Moshyfawn** [comment]
This is more in the realm of a discussion, so I'm going to convert it into one.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
